python/mxnet/io.py

```diff
@@ -95,6 +96,8 @@ def __init__(self, data, label, pad=None, index=None,
         self.bucket_key = bucket_key
         self.provide_data = provide_data
         self.provide_label = provide_label
+        self.provide_data_type = provide_data_type
```

Comment:
please remove these and merge them into provide_data/label with DataDesc
provide_data = [DataDesc(name, shape) ...]